commit 8ce243125c055a07f311492813aab4694e4732ca (refs/changes/86/4586/1)
Author: Jared Morrow <jm@joyent.com>
Date:   2018-07-26T10:03:46-06:00 (1 year, 2 months ago)
    
    MANTA-3774 Use fastdelete method to delete objects if user account has snaplinks disabled

diff --git a/lib/obj.js b/lib/obj.js
index 4a5233a..7d82bb9 100644
--- a/lib/obj.js
+++ b/lib/obj.js
@@ -926,6 +926,22 @@ function deletePointer(req, res, next) {
     };
 
     log.debug(opts, 'deletePointer: entered');
+
+    /*
+     * Let the delete mechanism know that snaplinks are disabled for this
+     * account, so it can treat the object differently
+     */
+    var uuid = req.owner.account.uuid;
+    common.checkAccountSnaplinksEnabled(req, uuid, function (enabled) {
+        // Pass in a special header if they are not enabled
+        if (!enabled) {
+            log.debug({
+                link: req.link,
+                owner: req.owner.account
+            }, 'deletePointer: owner of object has snaplinks disabled');
+            opts.snapLinksDisabled = true;
+        }
+    });
     req.moray.delMetadata(opts, function (err) {
         if (err) {
             next(err);
diff --git a/package.json b/package.json
index 0b04a23..b166b56 100644
--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
         "kang": "1.1.0",
         "keep-alive-agent": "0.0.1",
         "keyapi": "git+https://github.com/joyent/keyapi.git#e14b3d58",
-        "libmanta": "git+https://github.com/joyent/node-libmanta.git#master",
+        "libmanta": "git+https://github.com/joyent/node-libmanta.git#dev-MANTA-3774",
         "libuuid": "0.1.2",
         "lru-cache": "2.3.1",
         "lstream": "0.0.4",
